django_lightweight_queue/management/commands/queue_runner.py

```diff
@@ -1,56 +1,35 @@
 import logging
-import optparse
 import daemonize
 
 from django.apps import apps
-from django.core.management.base import CommandError, NoArgsCommand
+from django.core.management.base import BaseCommand
 
-from ...utils import get_backend, get_middleware, configure_logging
+from ...utils import get_backend, get_middleware, load_extra_config, configure_logging
 from ...runner import runner
 
-class Command(NoArgsCommand):
-    option_list = NoArgsCommand.option_list + (
-        optparse.make_option('--pidfile', action='store', dest='pidfile', default=None,
-            help="Fork and write pidfile to this file."),
-        optparse.make_option('--logfile', action='store', dest='logfile', default=None,
-            help="Log to the specified file."),
-        optparse.make_option('--touchfile', action='store', dest='touchfile', default=None,
-            help="touch(1) the specified file after running a job."),
-        optparse.make_option('--machine', action='store', dest='machine_number', default='1',
-            help="Machine number, for parallelism"),
-        optparse.make_option('--of', action='store', dest='machine_count', default='1',
-            help="Total number of machines running the queues"),
-        optparse.make_option('--only-queue', action='store', default=None,
-            help="Only run the given queue, useful for local debugging"),
-        optparse.make_option('--num-queue-workers', action='store', default=None,
-            help="The number of queue workers to run, only valid when running a specific queue"),
-    )
-
-    def handle_noargs(self, **options):
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('--pidfile', action='store', dest='pidfile', default=None,
+            help="Fork and write pidfile to this file.")
+        parser.add_argument('--logfile', action='store', dest='logfile', default=None,
+            help="Log to the specified file.")
+        parser.add_argument('--touchfile', action='store', dest='touchfile', default=None,
+            help="touch(1) the specified file after running a job.")
+        parser.add_argument('--machine', action='store', dest='machine_number', default='1',
+            help="Machine number, for parallelism")
+        parser.add_argument('--of', action='store', dest='machine_count', default='1',
+            help="Total number of machines running the queues")
+        parser.add_argument('--only-queue', action='store', default=None,
+            help="Only run the given queue, useful for local debugging")
+        parser.add_argument('--config', action='store', default=None,
+            help="The path to an additional django-style config file to load")
```

Comment:
Given that we're introducing a more powerful (and IMO clearer) mechanism, I'd much rather remove the feature as it would (now) violates the idea of having exactly one way to to something.

I've also just realised that it would have potentially confusing interactions with `--of` since the runner will run a fraction of the workers you might expect; this is also made much clearer by the `--exact-configuration` mechanism in #15.

(Frankly I'd argue that we could drop `--only-queue` in #15 too!)